packages/node/src/sdk.ts

```diff
@@ -180,3 +206,22 @@ export function getSentryRelease(fallback?: string): string | undefined {
     fallback
   );
 }
+
+/**
+ * Enable automatic Session Tracking for the node process.
+ */
+function startSessionTracking(): void {
+  const hub = getCurrentHub();
+  hub.startSession();
+  // Emitted in the case of healthy sessions, error of `mechanism.handled: true` and unhandledrejections because
+  // The 'beforeExit' event is not emitted for conditions causing explicit termination,
+  // such as calling process.exit() or uncaught exceptions.
+  // Ref: https://nodejs.org/api/process.html#process_event_beforeexit
+  process.on('beforeExit', () => {
+    const session = hub.getScope()?.getSession();
+    const terminalStates = [SessionStatus.Exited, SessionStatus.Crashed];
+    // Only call endSession, if the Session exists on Scope and SessionStatus is not a
+    // Terminal Status i.e. Exited or Crashed
```

Comment:
nit: Let's explain why we do this here.
If I understood correctly it's because if a session is in that state we've already sent the session to sentry and we don't want to do it twice.